File: python/cooker.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

import threading
import importlib;
from cookbook.util.logging import LoggerFactory

logger = LoggerFactory.getLogger(__name__)

def cook(recipes):

    for recipe in recipes:

        recipe_mod = importlib.import_module(recipe)

        logger.debug('')
        recipe_mod.title()
        recipe_mod.cook()

if __name__ == '__main__':

    threading.currentThread().name = 'main'

    recipes = [
        'cookbook.recipe_base',
        'cookbook.recipe_loop',
        'cookbook.recipe_command',
        'cookbook.recipe_thread'
    ]

    cook(recipes)

File: ruby/cooker.rb
#!/usr/bin/env ruby

require './cookbook/util/logging'

def cook recipes

    logger = LoggerFactory.getLogger File.basename __FILE__

    recipes.each do |recipe|

        require recipe

        logger.debug ''
        Recipe.title
        Recipe.cook

    end

end

if __FILE__ == $0

    Thread.current[:name] = 'main'

    recipes = [
        './cookbook/recipe_base',
        './cookbook/recipe_loop',
        './cookbook/recipe_command',
        './cookbook/recipe_thread'
    ]

    cook recipes

end
